templates/AWSServiceRoleForEC2Spot.yaml

```diff
@@ -0,0 +1,18 @@
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  AWSServiceRoleForEC2Spot:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: AWSServiceRoleForEC2Spot
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - spot.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: "/"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/aws-service-role/AWSEC2SpotServiceRolePolicy
```

Comment:
i assume you will be using this in one of your other templates so you should add an `Outputs:` section to allow cross stack references.  In the ref section: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html

you'll want to output resource Name, ARN and RoleId

Reply:
No, I will not be using this in any of my templates. This is a role that needs to exist so that spot instances can be created. It is one of those default service-linked roles that is required by some AWS services, like spot and EMR.

Reply:
Here is the doc that describes this: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#service-linked-roles-spot-instance-requests

Reply:
Ugh! Seriously AWS? I see why you think it should be in bridge-infra now.  I think this should be in our global infrastructure templates. Lemme see if I can make that happen